File: Qytetet_el_juego/p1yp2/QytetetJava/src/modeloqytetet/Jugador.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

import java.util.ArrayList;

/**
 *
 * @author chipi
 */
public class Jugador {
    
    private boolean encarcelado;
    private String nombre;
    private int saldo;
    private Sorpresa cartaLibertad; //0..1 cartaLibertad.gettipo()==TipoSorpresa.SalirCarcel
    private Casilla casillaActual;
    private ArrayList<TituloPropiedad> propiedades;

    public Jugador(String nombre) {
        this.nombre = nombre;
        this.encarcelado = false ;
        this.saldo = 7500 ;
        this.cartaLibertad= new Sorpresa("sorpresa",0,TipoSorpresa.SALIRCARCEL );
        this.casillaActual = new Casilla(0,0,TipoCasilla.SALIDA);
        this.propiedades = new ArrayList();
    }
    
    
    
    public Casilla getCasillaActual(){
        return this.casillaActual;
    }
    public boolean getEncarcelado(){
        return this.encarcelado;
    }
    /*
    public boolean tengoPropiedades(){
        return true;
    }
    boolean actualizarPosicion(Casilla casilla){
        return true;
    }
    boolean comprarTitulo(){
        return true;
    }
     Sorpresa devolverCartaLibertad(){
        return null;
    }
    void irACarcel(Casilla casilla){
            //casilla = carcel
    }
    void modificarSaldo(int cantidad ){

    }
    int obtenerCapital(){
        return 0;
    }
    ArrayList<TituloPropiedad> obtenerPropiedadesHipotecadas(boolean hipotecada) {
        return null;
    }
    void pagarCobrarPorCasaYHotel(int cantidad){

    }
    
    boolean pagarLibertad(int cantidad){ // cantidad = PrecioLibertad
        return true;
    }
    
    boolean puedoEdificarCasa(Casilla casilla){
        return true;
    }
    boolean puedoEdificarHotel(Casilla casilla){
        return true;
    }
    boolean puedoHipotecar(Casilla casilla){
        return true;
    }
    boolean puedoPagarHipoteca(Casilla casilla){
        return true;
    }
    boolean puedoVenderPropiedad(Casilla casillaa){
        return true;
    }
    */

    void setCartaLibertad(Sorpresa carta){ //carta : Sorpresa = cartaActual
        this.cartaLibertad = carta ;
    }
    
    void setCasillaActual(Casilla casilla){
        this.casillaActual = casilla ;
    }
    
    void setEncarcelado(boolean encarcelado ){
        this.encarcelado = encarcelado;
    }
    /*
    boolean tengoCartaLibertad(){
        return true;
    }
    void venderPropiedad(Casilla casilla){
    
    }
    
    private int cuantasCasasHotelesTengo(){
        return 0;
    }
    private void eliminarDeMisPropiedades(Casilla casilla){
    
    }
    
    private boolean esDeMipropiedad(Casilla casilla){
        return true;
    }
    
    private boolean tengoSaldo(int cantidad){
        return true;
    }

    */

    @Override
    public String toString() {
        return "Jugador{" + "\n encarcelado=" + encarcelado + 
                ",\n nombre=" + nombre + 
                ",\n saldo=" + saldo + 
                ",\n cartaLibertad=" + cartaLibertad + 
                ",\n casillaActual=" + casillaActual + 
                ",\n propiedades=" + propiedades + '}';
    }
    
    
}

File: Examenes/examen1/QytetetRuby/lib/jugador.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "casilla"
require_relative "billete"
require_relative "tipo_billete"

module ModeloQytetet
  class Jugador
    
    attr_accessor :encarcelado, :nombre, :saldo, :cartaLibertad, :casillaActual, :propiedades, :misBilletes
    
    def initialize(nombre,billetes)
      @encarcelado = false #boolean
      @nombre = nombre #string
      @saldo = 7500 #int
      @cartaLibertad #= Sorpresa.new('sorpresa',0,TipoSorpresa::salircarcel)
      @casillaActual #= Casilla.new(0,0,TipoCasilla::salida)
      @propiedades = Array.new()
      @misBilletes = billetes
    end
    
    def self.crear(nombre)
      new(nombre, Array.new);
    end
    
    def consultarSaldoBilletes()#int
      salida = 0 ;
      @misBilletes.each do |bil|  
        if(bil.tipo == TipoBillete::DE10)
        salida = salida + 10
        else
          if(bil.tipo == TipoBillete::DE50)
            salida = salida + 50
          else
            if(bil.tipo == TipoBillete::DE100) # no necesario, pero para saber que no hay basura en el array
              salida = salida + 100
            end
          end
        end
      end
      return salida;
    end
    
    def nuevoBillete(billetazo)
      @misBilletes << billetazo
    end
    
    def tengoPropiedades #public boolean tengoPropiedades()
      raise ArgumentError,'metodo sin implementar'
    end
    
    def actualizarPosicion(casilla) #boolean actualizarPosicion(Casilla casilla){
      raise ArgumentError,'metodo sin implementar'
    end
    
    def comprarTitulo #boolean comprarTitulo()
      raise ArgumentError,'metodo sin implementar'
    end
    
    def devolverCartaLibertad #Sorpresa devolverCartaLibertad()
      raise ArgumentError,'metodo sin implementar'
    end
    
    def irACarcel(casilla) #void irACarcel(Casilla casilla){
            #casilla = carcel
      raise ArgumentError,'metodo sin implementar'
    end
    
    def modificarSaldo(cantidad) #void modificarSaldo(int cantidad )
      raise ArgumentError,'metodo sin implementar'
    end
    
    def obtenerCapital #int obtenerCapital()
      raise ArgumentError,'metodo sin implementar'
    end
    
    def obtenerPropiedadesHipotecadas(hipotecada) #ArrayList<TituloPropiedad> obtenerPropiedadesHipotecadas(boolean hipotecada) {
      raise ArgumentError,'metodo sin implementar'
    end
    
    def pagarCobrarPorCasaYHotel(cantidad) #void pagarCobrarPorCasaYHotel(int cantidad)
      raise ArgumentError,'metodo sin implementar'
    end
    
    def pagarLibertad(cantidad) #boolean pagarLibertad(int cantidad){ // cantidad = PrecioLibertad
      raise ArgumentError,'metodo sin implementar'
    end
    
    def puedoEdificarCasa(casilla)#boolean puedoEdificarCasa(Casilla casilla){
      raise ArgumentError,'metodo sin implementar'
    end
    
    def puedoEdificarHotel(casilla)#boolean puedoEdificarHotel(Casilla casilla){
      raise ArgumentError,'metodo sin implementar'
    end
    
    def puedoHipotecar(casilla)#boolean puedoHipotecar(Casilla casilla){
      raise ArgumentError,'metodo sin implementar'
    end
    
    def puedoPagarHipoteca(casilla)#boolean puedoPagarHipoteca(Casilla casilla){
      raise ArgumentError,'metodo sin implementar'
    end
    
    def puedoVenderPropiedad(casilla)#boolean puedoVenderPropiedad(Casilla casillaa){
      raise ArgumentError,'metodo sin implementar'
    end
    
    def tengoCartaLibertad#boolean tengoCartaLibertad()
      raise ArgumentError,'metodo sin implementar'
    end
    
    def venderPropiedad(casilla)#void venderPropiedad(Casilla casilla)
      raise ArgumentError,'metodo sin implementar'
    end
    
    private
    
    def cuantasCasasHotelesTengo #private int cuantasCasasHotelesTengo()
      raise ArgumentError,'metodo sin implementar'
    end
    
    def eliminarDeMisPropiedades(casilla) #private void eliminarDeMisPropiedades(Casilla casilla)
      raise ArgumentError,'metodo sin implementar'
    end
    
    def esDeMipropiedad(casilla) #private boolean esDeMipropiedad(Casilla casilla)
      raise ArgumentError,'metodo sin implementar'
    end
    
    def tengoSaldo(cantidad) #private boolean tengoSaldo(int cantidad)
      raise ArgumentError,'metodo sin implementar'
    end 
    
    public 
    
    def to_s
      
      if(@misBilletes != nil)
        
        "Jugador: \n
      JugadorEncarcelado: #{@encarcelado}\n
      Saldo: #{@saldo} \n 
      Nombre: #{@nombre} \n 
      CartaLibertad: #{@cartaLibertad} \n
      CasillaActual: #{@casillaActual} \n 
      Propiedades: #{@propiedades} \n 
      misBilletes: #{@misBilletes} "
      else
        "Jugador: \n
      JugadorEncarcelado: #{@encarcelado}\n
      Saldo: #{@saldo} \n 
      Nombre: #{@nombre} \n 
      CartaLibertad: #{@cartaLibertad} \n
      CasillaActual: #{@casillaActual} \n 
      Propiedades: #{@propiedades} \n 
      misBilletes: no tiene billetes"
      end
    end
    
  end
end